UnityProject/Assets/Scripts/Managers/PlayerList.Admin.cs

```diff
@@ -211,7 +246,23 @@ private async Task<bool> CheckUserState(string userid, string token, ConnectedPl
 			return false;
 		}
 
+		//whitelist checking:
+		//Checks whether the userid is in either the Admins or whitelist or is server AND that the whitelist file has something in it.
+		//Whitelist only activates if whitelist is populated.
+
+		var lines = File.ReadAllLines(whiteListPath);
+
+		if ((!whiteListUsers.Contains(userid) | !adminUsers.Contains(userid)) && CustomNetworkManager.Instance._isServer == false && lines.Length > 0)
```

Comment:
`if ((!whiteListUsers.Contains(userid) | !adminUsers.Contains(userid)) && CustomNetworkManager.Instance._isServer == false && lines.Length > 0)
`
So no need to worry about this check `CustomNetworkManager.Instance._isServer ` as its already the server you are running the code on. 

change this `(!whiteListUsers.Contains(userid) | !adminUsers.Contains(userid))` too

`(!whiteListUsers.Contains(userid) && !adminUsers.Contains(userid)` So if you are on the whitelist or if you are on the admin list it won't trigger the kick